### ISSUE TITLE
Updating to 12.0.0 causes Jest to throw DND Draggable warnings

### ISSUE THREAD

**Issue by mattcarlotta** [atlassian/react-beautiful-dnd#1593]
### Expected behavior

No warnings will be thrown.

### Actual behavior

When invoking a class method within Jest to simulate the `onDragEnd` callback, this causes DND to throw warnings:

```
Unable to find any drag handles in the context "0"
```

```
A setup problem was encountered.
Invariant failed: Draggable[id: 5d72dffe65ec39141ae78553]: Unable to find drag handle
```

### Steps to reproduce

1. In Jest, invoke a class method that handles the result of a drag and drop.
2. View console to see warnings.

### Suggested solution?

Suppress warnings when in a test environment.

### What version of `React` are you using?

16.8.6 (also tested on 16.11.0)

### What version of `react-beautiful-dnd` are you running?

12.0.0

### What browser are you using?

Chrome Version 78.0.3904.97 (Official Build) (64-bit)

### Demo

[![Edit React DND - Warnings](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/react-dnd-moving-between-lists-array-1tj12?fontsize=14)

Click the `Simulate Jest Test` button to see that warnings aren't thrown.
Click on the `Tests` tab to run the tests. Look at the console to see DND warnings.

**Comment by alexreardon** [comment]
This is strange - we run all of our tests in Jest

**Comment by mattcarlotta** [comment]
I agree. Unfortunately, this isn't confined to the codesandbox either. I'm able to replicate it in a standalone application I'm working on -- only in a test environment.

**Comment by alexreardon** [comment]
Interesting. I'll have a play too

**Comment by alexreardon** [comment]
It looks like an enzyme setup issue.

I was able to fix the issue by adding these lines:

```diff
let wrapper;
  let root;
  beforeEach(() => {
    root = document.createElement("div");
    root.id = "root";
    document.body.appendChild(root);
    wrapper = mount(<Schedule {...initProps} />, { attachTo: root });
  });

  afterEach(() => {
    document.body.removeChild(root);
  });
```

I was doing some console.log statements inside a test and found that the document.body was strangely unpopulated. Making it attach to the `root` seemed to fix it up.


https://codesandbox.io/s/react-dnd-warnings-feyyt

**Comment by alexreardon** [comment]
Let me know if you want to reopen this

**Comment by mattcarlotta** [comment]
Thanks for the help. I'm still seeing the warnings when running the tests in your codesandbox, however changing the `afterEach` to an `afterAll` seemed to have fixed the warnings in the forked [codesandbox](https://codesandbox.io/s/react-dnd-warnings-1xbhb). I'll try testing it out in my web application and report back. In the meantime, if this is an enzyme only problem, I'll poke them for a fix.

**Comment by mattcarlotta** [comment]
The above seems to have fixed the issue! Thanks for the help!

**Comment by rpearce** [comment]
Confirmed this exact same issue with using `enzyme`'s `mount` and that https://github.com/atlassian/react-beautiful-dnd/issues/1593#issuecomment-552351855 fixes it. :+1:

**Comment by rpearce** [comment]
@alexreardon It seems the issue still arises if you `mount` twice in the same file (the example above has the `mount` done 1 time in a `beforeEach`).

Is this something worth looking in to, or would you rather wait for more folks to make noise about it?

**Comment by alexreardon** [comment]
PR's welcome @rpearce!

**Comment by rpearce** [comment]
I gave up and switched to testing r-b-dnd with `@testing-library/react`. No problems!

**Comment by renaudtertrais** [comment]
I fix the same issue by just [removing the warnings](https://github.com/atlassian/react-beautiful-dnd/blob/4721a518356f72f1dac45b5fd4ee9d466aa2996b/docs/guides/setup-problem-detection-and-error-recovery.md#disable-logging) with:
```js
// disable all react-beautiful-dnd development warnings
window['__react-beautiful-dnd-disable-dev-warnings'] = true;
```
Thoses warnings didn't have any impact on my tests result.

**Comment by akshaykharche** [comment]
> If those warnings doesn't impact your tests result, perhaps just [remove them](https://github.com/atlassian/react-beautiful-dnd/blob/4721a518356f72f1dac45b5fd4ee9d466aa2996b/docs/guides/setup-problem-detection-and-error-recovery.md#disable-logging):
> 
> ```js
> // disable all react-beautiful-dnd development warnings
> window['__react-beautiful-dnd-disable-dev-warnings'] = true;
> ```

Not sure where to apply above lines of code.Could you please elaborate it.

**Comment by matt-leyrat** [comment]
Howdy! I ran into this issue as well (`rbd v13.0.0`, `jest-enzyme ^7.1.2`), and this [Stackoverflow answer](https://stackoverflow.com/questions/53426148/mocking-react-beautiful-dnd-with-jest) solved it for me.

Adding this snippet to the top of your test file should get rid of the warnings. You can also follow [this doc](https://jestjs.io/docs/en/manual-mocks) to mock it globally.

```
jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
  Draggable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
  DragDropContext: ({ children }) => children,
}));
```

**Comment by asok-mirror** [comment]
I am still having this issue with Jest.
